nodejs/awsx/elasticloadbalancingv2/network.ts

```diff
@@ -133,16 +133,16 @@ export class NetworkListener
         this.registerOutputs({});
     }
 
-    public target(name: string, parent: pulumi.Resource): pulumi.Input<x.apigateway.ProxyTarget> {
+    public target(name: string, parent: pulumi.Resource): pulumi.Input<x.apigateway.IntegrationTarget> {
         // create a VpcLink to the load balancer in the VPC
         const vpcLink = new aws.apigateway.VpcLink(name, {
             targetArn: this.loadBalancer.loadBalancer.arn,
         }, { parent });
 
         return this.endpoint.apply(ep => ({
             uri: `http://${ep.hostname}:${ep.port}/`,
-            type: <x.apigateway.ApigatewayIntegrationType>"http_proxy",
-            connectionType: <x.apigateway.ApigatewayConnectionType>"VPC_LINK",
+            type: <x.apigateway.IntegrationType>"http_proxy",
```

Comment:
Are these casts necessary? Seems to me that `IntegrationType` is a union that includes `"http_proxy"`.

Reply:
unfortunately yes.   the problem here is that TS infers "string" as the type, not the explicit string-type 'http_proxy'.  This is a known issue with TS that they're considerign improving on for the future.